### PR TITLE
feat: support config from cli args and config file

### LINKED ISSUE #55: Add support for config js file and cli args
**Is your feature request related to a problem? Please describe.**

Currently all config options are specified using environment variables.

**Describe the solution you'd like**

Support a `commit-watch.config.js` file and/or cli args to specify all options.

**Describe alternatives you've considered**

A ".json" file could work too, but can't be dynamically updated for "CI". Not worth supporting 4 different config methods.

**Additional context**

Env variables are messy, hard to set, and leak all over the place.

### PR DESCRIPTION
**Pull request by AnvarGazizovTH**
## Description

Add ability to use file for config as well as command line arguments.

To use specify a config file use the `--config-file` cli argument to pass in the relative path of the file.

To pass in any specific config parameter as a command line argument use the parameter name as a cli argument.
ex. `commit-watch --github-token  password123` 

If config parameters are present from multiple sources they are used in the following order:
1. Configs passed in as cli args
2. Configs from a config file
3. Environment variable configs

## Related Issues

- Closes #55, adds support for a config JS file and CLI args.

## Checklist

- [x] This PR has sufficient documentation.
- [x] This PR has sufficient test coverage.
- [x] This PR title satisfies semantic [convention](https://www.conventionalcommits.org/en/v1.0.0/#summary).

### CODE REVIEW

File: src/getConfig.js [hunk 1]
```diff
@@ -1,8 +1,48 @@
+import minimist from 'minimist'
+
 import getCIVars from './getCIVars'
 import ensureValid from './ensureValid'
+import { parseConfigFile } from './parseConfigFile'
+
+const SUPPORTED_CONFIGS = [
+    'COMMITWATCH_GITHUB_TOKEN',
```
Comment (noahnu) [inline src/getConfig.js:8]: The "COMMITWATCH" and "COMMIT_WATCH" prefixes are redundant.

File: src/parseConfigFile.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import path from 'path'
+
+export const parseConfigFile = configFileRelativePath => {
+    const absolutePath = path.resolve(configFileRelativePath)
+    return require(absolutePath)
```
Comment (noahnu) [inline src/parseConfigFile.js:5]: I wonder if this works with yarn berry. We were seeing issues with lerna's similar usage of require.

File: README.md [hunk 3]
```diff
@@ -44,6 +44,23 @@ Before running commit-watch, you need to ensure the relevant environment variabl
 | COMMIT_WATCH_OUTPUT_FILENAME | Defaults to `commitwatch.junit.xml`.                 | The name of the junit report.                                                                        |
 | VERBOSE                      | Defaults to `0`.                                     | Whether to enable verbose mode.                                                                      |
 
+These variables can be set in the folowing ways: 
+1. By having environment variables with matching names.
+2. By specifying a config file containing the variables when running the command.
```
Comment (noahnu) [inline README.md:49]: let's include an example config in the readme

File: src/getConfig.js [hunk 4]
```diff
@@ -1,8 +1,48 @@
...
+    'CI_REPO_OWNER',
+    'CI_REPO_NAME',
+    'CI_COMMIT_SHA',
+    'CI_BASE_BRANCH',
+    'COMMIT_WATCH_OUTPUT_DIR',
+    'COMMIT_WATCH_OUTPUT_FILENAME',
+    'VERBOSE',
```
Comment (noahnu) [inline src/getConfig.js:15]: Let's follow standard convention of lowercase, hyphen separated words for the flags. E.g.

```
yarn commit-watch --verbose --base-branch origin/master --repo-name tophat
```

etc...

File: src/parseConfigFile.js (hunk 2 above)
Comment (AnvarGazizovTH) [reply src/parseConfigFile.js:5]: Not sure about this, what is a good alternative here?

Comment (noahnu) [reply src/parseConfigFile.js:5]: For yarn berry support: `require.resolve` is your friend. Here's an example: https://github.com/tophat/monodeploy/blob/7bb1e2165609df9bcd7ca75cd992d358a8315c84/src/utils/versionStrategy.ts#L72

File: README.md [hunk 5]
```diff
@@ -44,6 +44,42 @@ Before running commit-watch, you need to ensure the relevant environment variabl
...
+1. By having environment variables with matching names.
+2. By specifying a .js config file containing the variables when running the command.
+3. By specifying the variables in the command line arguments when running the command. 
+
+To use specify a config file use the --config-file cli argument to pass in the relative path of the file.
+
+ex. commit-watch --config-file commit-watch.config.js     
+    
+    
+    //commit-watch.coinfig.js
```
Comment (noahnu) [inline README.md:57]:
```suggestion
    //commit-watch.config.js
```

File: README.md [hunk 6]
```diff
@@ -44,6 +44,42 @@ Before running commit-watch, you need to ensure the relevant environment variabl
...
+    
+	
+	module.exports = {
+	    CI_REPO_OWNER: 'Hans Moleman',
+	    CI_REPO_NAME: 'manahattan-project',
+	    CI_BASE_BRANCH: 'develop',
+	    OUTPUT_DIR: 'junit_reports',
+	    OUTPUT_FILENAME: 'myreport.juint.xml',
+	    VERBOSE: true,
+	}
```
Comment (noahnu) [inline README.md:67]: Thoughts on camel case so it conforms more to eslint and jest config conventions?

**Comment by noahnu** [comment]
@AnvarGazizovTH  Left some comments. You'll also want to rebase on master.

File: README.md [hunk 7]
```diff
@@ -31,19 +31,57 @@ Or just run it directly in CI via `npx commit-watch` or `yarn dlx commit-watch`.
...
+
+To pass in any specific config parameter as a command line argument use the parameter name as a cli argument.
+
+ex. commit-watch --github-token password123
+
+If config parameters are present from multiple sources they are used in the following order:
+
+1. Configs passed in as cli args
+2. Configs from a config file
+3. Environment variable configs
```
Comment (noahnu) [inline README.md:82]: I wonder if env vars should take precedence over a config file? Not 100% sure. How do other tools handle this?

Comment (WulffHunter) [reply README.md:82]: I found this on StackOverflow: https://stackoverflow.com/questions/11077223/what-order-of-reading-configuration-values#:~:text=If%20a%20config%20file%20is,will%20take%20precedence%20over%20it.

**Review by noahnu — APPROVED** [review]

**Comment by noahnu** [comment]
@all-contributors add @WulffHunter for code

**Pull request merged**